runtime/tests/org.eclipse.core.tests.runtime/src/org/eclipse/core/tests/runtime/jobs/IJobManagerTest.java

```diff
@@ -1750,33 +1773,15 @@ protected IStatus run(IProgressMonitor monitor) {
 		}
 	}
 
-	public void testReverseOrder() {
-		//ensure jobs are run in order from lowest to highest sleep time.
-		final Queue<Job> done = new ConcurrentLinkedQueue<>();
-		int[] sleepTimes = new int[] { 600, 400, 200, 5 };
-		Job[] jobs = new Job[sleepTimes.length];
-		for (int i = 0; i < sleepTimes.length; i++) {
-			jobs[i] = new Job("testReverseOrder(" + i + ")") {
-
-				@Override
-				protected IStatus run(IProgressMonitor monitor) {
-					done.add(this);
-					return Status.OK_STATUS;
-				}
-
-			};
-		}
-		for (int i = 0; i < sleepTimes.length; i++) {
-			jobs[i].schedule(sleepTimes[i]);
-		}
-		while (done.size() != jobs.length) {
-			Thread.yield();
-		}
-		Job[] doneOrder = done.toArray(new Job[done.size()]);
-		assertEquals("1.0", jobs.length, doneOrder.length);
-		for (int i = 0; i < doneOrder.length; i++) {
-			assertEquals("1.1." + i, jobs[i], doneOrder[doneOrder.length - 1 - i]);
-		}
+	public void testReverseOrder() throws InterruptedException {
+		// ensure that a job does not wait for one that is scheduled with a high delay
+		Job directlyExecutedJob = new TestJob("Directly executed job", 0, 0);
+		Job delayedJob = new TestJob("Delayed job", 0, 0);
```

Comment:
that job should end before tests end to not interfere with other tests. cancel it for example

Reply:
Yes, thanks! The `cancel` call was missing. Fixed in 890973e.